app/tests/conftest.py

```
@@ -2,26 +2,45 @@
+    app.dependency_overrides[get_email_service] = lambda: svc
+    yield svc
+    app.dependency_overrides.pop(get_email_service, None)
+
 
 @pytest.fixture
-def client(vault_repo: InMemoryVaultRepository) -> Generator[TestClient, None, None]:
-    """
-    Fixture that provides a TestClient.
-    Depends on vault_repo to ensure overrides are in place.
-    """
+def client(app, vault_repo) -> Generator[TestClient, None, None]:
```

Ruff (0.14.14)

[warning] Unused function argument: `vault_repo`

(ARG001)